.github/workflows/format.yml

```diff
@@ -0,0 +1,23 @@
+name: Scalafmt
+
+permissions: {}
+
+on:
+  pull_request:
+    branches: ['**']
+
+jobs:
+  build:
+    name: Code is formatted
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout current branch (full)
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+
+      - name: Check project is formatted
+        uses: jrouly/scalafmt-native-action@v3
+        with:
+          arguments: '--list --mode diff-ref=origin/develop'
```

Comment:
Wouldn't this diff-ref delay the formatting of the code when Scalafmt version is changed, and end up creating spurious changes when the next person comes along to touch a Scala file?